devicetypes/smartthings/zigbee-thermostat.src/zigbee-thermostat.groovy

```diff
@@ -261,8 +260,8 @@ private parseAttrMessage(description) {
 				state.supportedFanModes = FAN_MODE_SEQUENCE_MAP[it.value]
 			}
 		} else if (it.cluster == zigbee.POWER_CONFIGURATION_CLUSTER) {
-			if (it.attribute == BATTERY_VOLTAGE) {
-				map = getBatteryPercentage(Integer.parseInt(it.value, 16))
+			if (it.attribute == BATTERY_VOLTAGE || it.attribute == BATTERY_PERCENTAGE_REMAINING) {
```

Comment:
handling of battery percentage attribute should either be done inline or in a separate function than the one that is expecting voltage measurments

Reply:
ok, done it inline, it should look better.